From 1bf698a99afd1bd50641696aee906ea322769186 Mon Sep 17 00:00:00 2001
From: Jinjiang Tu <tujinjiang@huawei.com>
Date: Mon, 15 Dec 2025 15:13:47 +0800
Subject: [PATCH 894/960] downstream: arm64: configs: enable UB memory pooling
 support

Enable MEMORY_HOTREMOVE, ZONE_EXTMEM, NUMA_REMOTE, PFN_RANGE_ALLOC,
RECLAIM_NOTIFY by default for arm64.
Enable huge pfnmap configurations for both arm64 and x86_64 architectures.

Signed-off-by: Jinjiang Tu <tujinjiang@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 config.aarch64 | 9 ++++++++-
 config.x86_64  | 3 +++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/config.aarch64 b/config.aarch64
index ff3e30e4063c..1d099db295bd 100644
--- a/config.aarch64
+++ b/config.aarch64
@@ -837,6 +837,7 @@ CONFIG_FUNCTION_ALIGNMENT=8
 
 CONFIG_RT_MUTEXES=y
 CONFIG_BASE_SMALL=0
+CONFIG_PFN_RANGE_ALLOC=y
 CONFIG_MODULE_SIG_FORMAT=y
 CONFIG_MODULES=y
 # CONFIG_MODULE_DEBUG is not set
@@ -1064,7 +1065,7 @@ CONFIG_ARCH_ENABLE_MEMORY_HOTPLUG=y
 CONFIG_ARCH_ENABLE_MEMORY_HOTREMOVE=y
 CONFIG_MEMORY_HOTPLUG=y
 CONFIG_MEMORY_HOTPLUG_DEFAULT_ONLINE=y
-# CONFIG_MEMORY_HOTREMOVE is not set
+CONFIG_MEMORY_HOTREMOVE=y
 CONFIG_MHP_MEMMAP_ON_MEMORY=y
 CONFIG_ARCH_MHP_MEMMAP_ON_MEMORY_ENABLE=y
 CONFIG_SPLIT_PTLOCK_CPUS=4
@@ -1092,6 +1093,9 @@ CONFIG_TRANSPARENT_HUGEPAGE=y
 CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
 CONFIG_THP_SWAP=y
 # CONFIG_READ_ONLY_THP_FOR_FS is not set
+CONFIG_ARCH_SUPPORTS_HUGE_PFNMAP=y
+CONFIG_ARCH_SUPPORTS_PMD_PFNMAP=y
+# CONFIG_ARCH_SUPPORTS_PUD_PFNMAP is not set
 CONFIG_NEED_PER_CPU_EMBED_FIRST_CHUNK=y
 CONFIG_NEED_PER_CPU_PAGE_FIRST_CHUNK=y
 CONFIG_USE_PERCPU_NUMA_NODE_ID=y
@@ -1110,6 +1114,8 @@ CONFIG_ARCH_HAS_PTE_DEVMAP=y
 CONFIG_ARCH_HAS_ZONE_DMA_SET=y
 CONFIG_ZONE_DMA=y
 CONFIG_ZONE_DMA32=y
+CONFIG_ZONE_EXTMEM=y
+CONFIG_NUMA_REMOTE=y
 CONFIG_HMM_MIRROR=y
 CONFIG_VM_EVENT_COUNTERS=y
 # CONFIG_PERCPU_STATS is not set
@@ -1120,6 +1126,7 @@ CONFIG_MEMFD_CREATE=y
 CONFIG_SECRETMEM=y
 # CONFIG_ANON_VMA_NAME is not set
 CONFIG_USERFAULTFD=y
+CONFIG_RECLAIM_NOTIFY=y
 CONFIG_HAVE_ARCH_USERFAULTFD_MINOR=y
 CONFIG_LRU_GEN=y
 # CONFIG_LRU_GEN_ENABLED is not set
diff --git a/config.x86_64 b/config.x86_64
index 56fed63d3e2c..2acb852690fc 100644
--- a/config.x86_64
+++ b/config.x86_64
@@ -1094,6 +1094,9 @@ CONFIG_TRANSPARENT_HUGEPAGE=y
 CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
 CONFIG_THP_SWAP=y
 # CONFIG_READ_ONLY_THP_FOR_FS is not set
+CONFIG_ARCH_SUPPORTS_HUGE_PFNMAP=y
+CONFIG_ARCH_SUPPORTS_PMD_PFNMAP=y
+# CONFIG_ARCH_SUPPORTS_PUD_PFNMAP is not set
 CONFIG_NEED_PER_CPU_EMBED_FIRST_CHUNK=y
 CONFIG_NEED_PER_CPU_PAGE_FIRST_CHUNK=y
 CONFIG_USE_PERCPU_NUMA_NODE_ID=y
-- 
2.43.0

